account/views.py

```diff
@@ -30,14 +30,21 @@
 account = Blueprint('account', __name__)
 
 
+def validate_login(form):
+    user = User.query.filter_by(email=form.email.data).first()
+    if user is not None and user.password_hash is not None and \
+            user.verify_password(form.password.data):
+        return user
+    else:
+        return None
+
 @account.route('/login', methods=['GET', 'POST'])
 def login():
     """Log in an existing user."""
     form = LoginForm()
     if form.validate_on_submit():
-        user = User.query.filter_by(email=form.email.data).first()
-        if user is not None and user.password_hash is not None and \
-                user.verify_password(form.password.data):
+        user = validate_login(form)
+        if user:
             login_user(user, form.remember_me.data)
```

Comment:
The login() function could be improved by separating the validation logic from the routing logic. This would make the code more modular and easier to test.